synapse/http/request_metrics.py

```diff
@@ -43,6 +43,24 @@
     "synapse_http_server_response_time_seconds",
     "sec",
     ["method", "servlet", "tag", "code"],
+    buckets=(
+        0.005,
+        0.01,
+        0.025,
+        0.05,
+        0.1,
+        0.25,
+        0.5,
+        1.0,
+        2.5,
+        5.0,
+        10.0,
+        30.0,
+        60.0,
+        120.0,
+        180.0,
+        "+Inf",
+    ),
```

Comment:
> Sorry for not jumping in on this sooner, but: the vast majority of our APIs are responding within the range of 0-10s, so losing fidelity there reduces our insights into response time. There is quite a big difference between APIs that return in 500ms and those that return in 1s, and removing the 750ms means we can't easily differentiate.
>
> Since this a thing that we're adding specifically to measure progress in performance improvements for a particular API, I'm very tempted to suggest that we simply create a separate metric for the `/messages` API. This would also allow us to differentiate between local and remote `/messages` requests for example.
>
> *-- @erikjohnston, https://github.com/matrix-org/synapse/pull/13478#issuecomment-1212976057*

I can create a separate PR to add a specific metric for `/messages` ⏩

But do we have any interest in adjusting the buckets for the general case? @erikjohnston mentioned if anything maybe wanting even more fidelity in the lower ranges. @richvdh do you have any interest in increasing for another endpoint? Our limiting factor is [cardinality](https://github.com/matrix-org/synapse/issues/11082) since this multiplies out to all of our servlets.

> I think @MadLittleMods is right in that the top bucket should be more than 10s given how often some of our endpoints take longer than that
>
> *-- @richvdh, https://matrix.to/#/!vcyiEtMVHIhWXcJAfl:sw1v.org/$CLJ5oioD_DO1A_zSGmYtCd-yToSyA6EiOwOsClvfdcs?via=matrix.org&via=element.io&via=beeper.com*

https://grafana.matrix.org/d/000000012/synapse?orgId=1&from=1660574475300&to=1660596075300&viewPanel=47

[![Slow `KeyChangesServlet`s shown in grafana](https://user-images.githubusercontent.com/558581/184713794-13118c2a-83cf-47c6-a8e1-26ffcc92e971.png)]( https://grafana.matrix.org/d/dYoRgTgVz/messages-timing?orgId=1&from=1660574188178&to=1660595788178&viewPanel=47)



---

In terms of reducing cardinality, we could remove `code`. I think for timing, we really just need the method and servlet name. Response `code` can be useful but maybe we just need to change it `successful_response` boolean (with a cardinality of 2, `[true|false]`) since we only ever use it as `code=~"2.."`

Reply:
>  @richvdh do you have any interest in increasing for another endpoint?

I'm not sure really. I guess? I don't feel like it's a huge priority to me right now.

> In terms of reducing cardinality, we could remove `code`.

Probably, yes. but that's a conversation we should have over at https://github.com/matrix-org/synapse/issues/11082.